### PR TITLE
Type safety at micrositeCompilingDocsTool Setting

### PR DESCRIPTION
**Pull request by juanpedromoreno**
This PR brings a new ADT that users can use at their convenience, with two possibilities: `tut` or `mdoc`.

### CODE REVIEW

File: src/main/scala/microsites/MicrositeKeys.scala [hunk 1]
```diff
@@ -28,8 +28,8 @@ import sbt.Keys._
 import sbt._
 import sbt.complete.DefaultParsers.OptNotSpace
 import sbtorgpolicies.github.GitHubOps
-import tut.TutPlugin.autoImport._
-import mdoc.MdocPlugin.autoImport._
+import tut.TutPlugin.autoImport.{tut => tutTask, _}
```
Comment (fedefernandez) [inline src/main/scala/microsites/MicrositeKeys.scala]: Do we need this TutPlugin autoImport and not the one for MdocImport?

File: src/main/scala/microsites/MicrositeKeys.scala [hunk 2]
```diff
@@ -43,6 +43,10 @@ trait MicrositeKeys {
   final case object GHPagesPlugin                  extends PushWith("ghPagesPlugin")
   final case object GitHub4s                       extends PushWith("github4s")
 
+  sealed abstract class CompilingDocsTool extends Product with Serializable
+  final case object tut                   extends CompilingDocsTool
+  final case object mdoc                  extends CompilingDocsTool
```
Comment (fedefernandez) [inline src/main/scala/microsites/MicrositeKeys.scala]: This is a bit redundant, right? I guess these can be just case object

**Comment by juanpedromoreno** [comment]
This is going to fail given `mdoc` and `tut` are reserved words when they are auto-triggered (`AutoPlugin`) where they are used :(

```bash
[error] it is imported twice in the same scope by
[error] import _root_.mdoc.MdocPlugin.autoImport._
[error] and import _root_.microsites.MicrositesPlugin.autoImport._
```

Therefore, we might have two options:

* Renaming `tut` and `mdoc` case objects by something else (`WithTut`, `WithMdoc`? 🤔 ).
* Closing this PR and leaving settings as it is (String values).

Thoughts?

**Comment by fedefernandez** [comment]
Will `Tut` and `MDoc` have the same problem?

**Comment by rafaparadela** [comment]
If, as Fede suggested, `Tut` and `Mdoc` work fine, seem to be the best choice. Otherwise `WithTut` and `WithMdoc` are not too bad either.

**Comment by juanpedromoreno** [comment]
`Tut` is also a reserved word in the tut side. I'll use `WithTut`, `WithMdoc`

**Pull request merged**